File: ref/python/clientthread.py
from messagefactory import MessageFactory
from socketthread import SocketThread
import messages


class ClientThread( SocketThread ):
	server = None
	send = None
	
	def __init__(self, socketAddr, server, logger=None):
		SocketThread.__init__(self,
							  socketAddr=socketAddr,
							  interface=server.interface,
							  logger=None,
							  recvmessage=messages.servermain,
							  sendmessage=messages.clientmain)
		self.server = server
		
		self.log( "Connected", level=4 )
		

File: templates/java/server/ClientThread.java
{% set package, java_subpackage = "server", "server" %}
package {{ java_package }};

import java.net.Socket;

import {{ java_package_root }}.common.Connection;
import {{ java_package_root }}.common.Messages;
import {{ java_package_root }}.common.SocketThread;

public class ClientThread extends SocketThread {

	public Server server;
	
	public ClientThread(Socket socket, Server server, Connection connection) {
		super(socket, server, connection, Messages.serverMain, Messages.clientMain);
		this.server = server;
	}
	
}
